drag_and_drop_v2/drag_and_drop_v2.py

```diff
@@ -457,11 +500,12 @@ def _is_finished(self):
         completed_count = 0
         total_count = 0
         item_state = self._get_item_state()
+
         for item in self.data['items']:
-            if item['zone'] != 'none':
+            item_id = item['id']
+            if self._get_valid_item_zones(item_id) != []:
                 total_count += 1
-                item_id = str(item['id'])
-                if item_id in item_state:
+                if str(item_id) in item_state:
                     completed_count += 1
```

Comment:
@haikuginger Aside from the return value, the logic in this method is identical to the logic in `_get_grade` above. It would be worth moving it to a separate method (something like `_get_item_stats`) I think. Then you could just call this method from `_get_grade` and `_is_finished` like this:

```python
correct_count, total_count = self._get_item_stats()
```